ui/pages/confirmations/components/confirm/info/shield-subscription-approve/subscription-details.tsx

```diff
@@ -73,7 +73,7 @@ export const SubscriptionDetails = ({
             }}
           >
             <Text variant={TextVariant.bodySm} color={TextColor.inherit}>
-              {t('freeSevenDayTrial')}
+              {t('freeTrialDays', [productPrice?.trialPeriodDays ?? ''])}
```

Comment:
we should fallback to a number greater than 1 here, so that the text still makes sense. Perhaps 7?